src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCopy.java

```diff
@@ -2167,7 +2167,8 @@ else if (null != sourceCryptoMeta) {
                         if (bulkNullable) {
                             tdsWriter.writeByte((byte) 0x08);
                         }
-                        tdsWriter.writeDouble((float) colValue);
+                        float floatValue = ((Number)colValue).floatValue();
+                        tdsWriter.writeDouble(floatValue);
```

Comment:
Please combine these lines as follows:
```suggestion
                        tdsWriter.writeDouble(((Number) colValue).floatValue());
```